backend/controllers/passController.js

```diff
@@ -55,14 +62,47 @@ const viewPasses = asyncHandler(async (req, res) => {
         message: "Passes fetched successfully",
       })
     } else if (passes.length < 1) {
-      res.status(201).json({
-        success: true,
-        message: "No passes found",
-      })
+      throw new Error("No passes found")
     }
   } catch (err) {
     console.log(err)
+    res.status(400).json(err?.message)
+  }
+})
+
+//@desc Verify pass by gaurd
+//@route PUT /users/verifyPass
+const verifyPass = asyncHandler(async (req, res) => {
+  try {
+    await connectDB()
+    const user = await User.findById(req.user._id)
+    console.log(req.params._id)
+
+    if (user.role == "gaurd") {
+      const passVerified = await Pass.findOneAndUpdate(
+        { _id: req.params._id },
+        {
+          checkInStatus: true,
+          timestamps: new Date(),
+        }
```

Comment:
So, Can you help me to understand a case in which 

1. `checkInStatus` is false but the `timestamp` variable is set with some time or
2. `checkInStatus` is true but the `timestamp` is undefined.

Reply:
1. when a user moves out of the organisation, after completing his work (in our case, suppose a washing machine serviceman comes inside, he will have some timeAtCheckIn which will be assigned and the checkInStatus will be true that time. When he returns back, his checkInStatus would be made false but he will still have the timeAtCheckIn being assigned to him when he exits)
2. When a non string value is passed for Date. Example, if undefined is the parameter of Date(), then it will be coerced to NaN which is an undefined time.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date

Reply:
1. Then are you sure about the variable name? CheckinStatus? 
2. Have you tried that? Is it storing the value?

Reply:
1. can we use something like userPermittedIn?? like, if the user is inside the org, its value would be true, and when outside (before going in and after completing his work he gets out) then the value can be false.
2. its storing "invalid date" when passed undefined in Date()

Reply:
1. Make sense. Will you need any other key to have some valuable information?
2. Okay.

Reply:
1. maybe I need to store a boolean value to "passUsed" or "userVisited" so as to check whether the pass has been used till now or not..

Reply:
also, should I now change "checkInStatus" to "userPermittedIn"?

Reply:
1. Okay, can't we interpret that by just checking `checkinTime`? Until the user is not checked In to the campus, the pass hasn't been used or hasn't been verified, Isn't it?

Reply:
yes, if the user has not been permitted into the campus, till then it should be marked as false, and when he wishes to enter then his pass can be marked as true..we can check the pass usage with that variable (initially i was thinking that the pass's checkInStatus could be marked back to false when the person goes out of the campus...)